parsl/executors/high_throughput/mac_safe_queue.py

```diff
@@ -0,0 +1,67 @@
+import multiprocessing
+import multiprocessing.queues
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class MacSafeQueue(multiprocessing.queues.Queue):
+    """ Multiprocessing queues do not have qsize attributes on MacOS.
+    This is slower but more portable version of the multiprocessing Queue
+    that adds a explicit counter
+
+    Reference : https://github.com/keras-team/autokeras/commit/4ddd568b06b4045ace777bc0fb7bc18573b85a75
+    """
+
+    def __init__(self, *args, **kwargs):
+        if 'ctx' not in kwargs:
+            kwargs['ctx'] = multiprocessing.get_context()
```

Comment:
multiprocessing.Queue is a method that does the extra step of context creation. I have to do this since I need the underlying class itself to inherit from. More info here : https://stackoverflow.com/questions/24941359/ctx-parameter-in-multiprocessing-queue